modules/expo-localization/android/src/main/java/expo/modules/localization/LocalizationModule.java

```diff
@@ -0,0 +1,187 @@
+package expo.modules.localization;
+
+import android.app.Activity;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.content.res.Configuration;
+import android.os.BaseBundle;
+import android.os.Build;
+import android.os.Bundle;
+import android.os.LocaleList;
+import android.text.TextUtils;
+import android.view.View;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.Currency;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.TimeZone;
+
+import expo.core.ExportedModule;
+import expo.core.ModuleRegistry;
+import expo.core.interfaces.ActivityProvider;
+import expo.core.interfaces.ModuleRegistryConsumer;
+import expo.core.interfaces.services.EventEmitter;
+
+import static android.content.Intent.ACTION_LOCALE_CHANGED;
+import static android.os.Build.VERSION.SDK_INT;
+import static android.os.Build.VERSION_CODES.LOLLIPOP;
+import static android.os.Build.VERSION_CODES.N;
+import static java.util.Currency.getAvailableCurrencies;
+
+public class LocalizationModule extends ExportedModule implements ModuleRegistryConsumer {
+
+    private LocalesBroadcastReceiver mReceiver;
+    private ModuleRegistry mModuleRegistry;
+    private EventEmitter mEventEmitter;
+
+    public LocalizationModule(Context context) {
+        super(context);
+    }
+
+    private final Context getApplicationContext() {
+        return getCurrentActivity().getApplicationContext();
+    }
+
+    private Activity getCurrentActivity() {
+        ActivityProvider activityProvider = mModuleRegistry.getModule(ActivityProvider.class);
+        return activityProvider.getCurrentActivity();
+    }
+
+    @Override
+    public String getName() {
+        return "ExpoLocalization";
+    }
+
+    @Override
+    public void setModuleRegistry(ModuleRegistry moduleRegistry) {
+
+        mModuleRegistry = moduleRegistry;
+        mEventEmitter = null;
+
+        if (mReceiver != null) {
+            getApplicationContext().unregisterReceiver(mReceiver);
+            mReceiver = null;
+        }
+
+        mEventEmitter = moduleRegistry.getModule(EventEmitter.class);
```

Comment:
Actually what I meant is that if `moduleRegistry == null` this and `getApplicationContext()` will throw a `NullPointerException` 😅